addons/mail/models/mail_message_schedule.py

```diff
@@ -63,7 +63,7 @@ def _send_notifications(self, default_notify_kwargs=None):
         """
         for model, schedules in self._group_by_model().items():
             if model:
-                records = self.env[model].browse(schedules.mapped('mail_message_id.res_id'))
+                records = self.env[model].browse(schedules.mapped('mail_message_id.res_id')).exists()
             else:
                 records = [self.env['mail.thread']] * len(schedules)
```

Comment:
Well no because then the schedule will be mismatched with the records won't they?

We need to somehow maintain the mapping between schedule and record. Maybe with a map from "schedule" to record, and if the schedule has no record I suppose we'd use an empty `mail.thread` to be consistent with the existing "else". So that the message still tries to send somehow, if it was originally valid (I guess that's the intent of the else case)